### ISSUE TITLE
about IOU

### ISSUE THREAD

**Issue by sunxia233** [silverbulletmdc/PVEN#23]
Thank you for your work. According to your suggestion, I trained after Convert polygons to parsing masks without modifying any code. This is my training log but IOU is not as high as in the paper.
(py36) ➜  parsing python train_parsing.py --train-set trainval --masks-path ../outputs/veri776_parsing3165 --image-path /media/soar/data/176/VeRi/image_train

Epoch: 0
train: 100%|████| 396/396 [00:55<00:00,  7.67it/s, bce_dice_loss - 1.606, iou_score - 0.2795]
valid: 100%|█████| 500/500 [00:09<00:00, 52.51it/s, bce_dice_loss - 1.519, iou_score - 0.434]
Model saved!

Epoch: 1
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.581, iou_score - 0.3631]
valid: 100%|████| 500/500 [00:09<00:00, 52.48it/s, bce_dice_loss - 1.502, iou_score - 0.5235]
Model saved!

Epoch: 2
train: 100%|████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.572, iou_score - 0.4105]
valid: 100%|████| 500/500 [00:09<00:00, 52.79it/s, bce_dice_loss - 1.493, iou_score - 0.5486]
Model saved!

Epoch: 3
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.564, iou_score - 0.4222]
valid: 100%|█████| 500/500 [00:09<00:00, 52.67it/s, bce_dice_loss - 1.49, iou_score - 0.5551]
Model saved!

Epoch: 4
train: 100%|████| 396/396 [00:56<00:00,  7.58it/s, bce_dice_loss - 1.563, iou_score - 0.4299]
valid: 100%|████| 500/500 [00:09<00:00, 52.67it/s, bce_dice_loss - 1.485, iou_score - 0.5562]
Model saved!

Epoch: 5
train: 100%|████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.555, iou_score - 0.4416]
valid: 100%|████| 500/500 [00:09<00:00, 52.69it/s, bce_dice_loss - 1.482, iou_score - 0.5473]

Epoch: 6
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.552, iou_score - 0.4419]
valid: 100%|████| 500/500 [00:09<00:00, 52.81it/s, bce_dice_loss - 1.474, iou_score - 0.5468]

Epoch: 7
train: 100%|████| 396/396 [00:56<00:00,  7.60it/s, bce_dice_loss - 1.545, iou_score - 0.4489]
valid: 100%|████| 500/500 [00:09<00:00, 52.79it/s, bce_dice_loss - 1.469, iou_score - 0.5649]
Model saved!

Epoch: 8
train: 100%|████| 396/396 [00:56<00:00,  7.58it/s, bce_dice_loss - 1.544, iou_score - 0.4505]
valid: 100%|████| 500/500 [00:09<00:00, 52.74it/s, bce_dice_loss - 1.467, iou_score - 0.5736]
Model saved!

Epoch: 9
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.536, iou_score - 0.4599]
valid: 100%|████| 500/500 [00:09<00:00, 52.66it/s, bce_dice_loss - 1.463, iou_score - 0.5841]
Model saved!

Epoch: 10
train: 100%|████| 396/396 [00:56<00:00,  7.60it/s, bce_dice_loss - 1.534, iou_score - 0.4581]
valid: 100%|████| 500/500 [00:09<00:00, 52.85it/s, bce_dice_loss - 1.459, iou_score - 0.5761]

Epoch: 11
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.531, iou_score - 0.4612]
valid: 100%|████| 500/500 [00:09<00:00, 52.90it/s, bce_dice_loss - 1.455, iou_score - 0.5709]

Epoch: 12
train: 100%|████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.525, iou_score - 0.4895]
valid: 100%|█████| 500/500 [00:09<00:00, 52.63it/s, bce_dice_loss - 1.451, iou_score - 0.607]
Model saved!

Epoch: 13
train: 100%|████| 396/396 [00:56<00:00,  7.60it/s, bce_dice_loss - 1.522, iou_score - 0.4973]
valid: 100%|████| 500/500 [00:09<00:00, 52.69it/s, bce_dice_loss - 1.446, iou_score - 0.6365]
Model saved!

Epoch: 14
train: 100%|█████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.52, iou_score - 0.5051]
valid: 100%|████| 500/500 [00:09<00:00, 52.71it/s, bce_dice_loss - 1.443, iou_score - 0.6174]

Epoch: 15
train: 100%|████| 396/396 [00:56<00:00,  7.58it/s, bce_dice_loss - 1.517, iou_score - 0.5044]
valid: 100%|████| 500/500 [00:09<00:00, 52.60it/s, bce_dice_loss - 1.442, iou_score - 0.6192]

Epoch: 16
train: 100%|████| 396/396 [00:56<00:00,  7.59it/s, bce_dice_loss - 1.514, iou_score - 0.5098]
valid: 100%|████| 500/500 [00:09<00:00, 52.46it/s, bce_dice_loss - 1.439, iou_score - 0.6245]

Epoch: 17
train: 100%|█████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.513, iou_score - 0.508]
valid: 100%|████| 500/500 [00:09<00:00, 52.99it/s, bce_dice_loss - 1.435, iou_score - 0.6242]

Epoch: 18
train: 100%|█████| 396/396 [00:56<00:00,  7.60it/s, bce_dice_loss - 1.511, iou_score - 0.507]
valid: 100%|████| 500/500 [00:09<00:00, 52.75it/s, bce_dice_loss - 1.432, iou_score - 0.6201]

Epoch: 19
train: 100%|████| 396/396 [00:56<00:00,  7.60it/s, bce_dice_loss - 1.504, iou_score - 0.5156]
valid: 100%|████| 500/500 [00:09<00:00, 52.69it/s, bce_dice_loss - 1.431, iou_score - 0.6176]

Epoch: 20
train: 100%|████| 396/396 [00:56<00:00,  7.60it/s, bce_dice_loss - 1.502, iou_score - 0.5199]
valid: 100%|████| 500/500 [00:09<00:00, 52.57it/s, bce_dice_loss - 1.429, iou_score - 0.6245]

Epoch: 21
train: 100%|████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.502, iou_score - 0.5173]
valid: 100%|████| 500/500 [00:09<00:00, 52.23it/s, bce_dice_loss - 1.426, iou_score - 0.6284]

Epoch: 22
train: 100%|██████| 396/396 [00:56<00:00,  7.63it/s, bce_dice_loss - 1.5, iou_score - 0.5171]
valid: 100%|████| 500/500 [00:09<00:00, 52.68it/s, bce_dice_loss - 1.423, iou_score - 0.6391]
Model saved!

Epoch: 23
train: 100%|████| 396/396 [00:56<00:00,  7.57it/s, bce_dice_loss - 1.494, iou_score - 0.5248]
valid: 100%|████| 500/500 [00:09<00:00, 52.82it/s, bce_dice_loss - 1.421, iou_score - 0.6225]

Epoch: 24
train: 100%|████| 396/396 [00:56<00:00,  7.57it/s, bce_dice_loss - 1.494, iou_score - 0.5213]
valid: 100%|██████| 500/500 [00:09<00:00, 52.59it/s, bce_dice_loss - 1.42, iou_score - 0.626]

Epoch: 25
train: 100%|████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.493, iou_score - 0.5225]
valid: 100%|████| 500/500 [00:09<00:00, 52.70it/s, bce_dice_loss - 1.417, iou_score - 0.6361]
Decrease decoder learning rate to 1e-5!

Epoch: 26
train: 100%|████| 396/396 [00:56<00:00,  7.63it/s, bce_dice_loss - 1.488, iou_score - 0.5318]
valid: 100%|████| 500/500 [00:09<00:00, 52.55it/s, bce_dice_loss - 1.416, iou_score - 0.6592]
Model saved!

Epoch: 27
train: 100%|█████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.49, iou_score - 0.5278]
valid: 100%|████| 500/500 [00:09<00:00, 52.98it/s, bce_dice_loss - 1.415, iou_score - 0.6351]

Epoch: 28
train: 100%|████| 396/396 [00:56<00:00,  7.58it/s, bce_dice_loss - 1.488, iou_score - 0.5313]
valid: 100%|████| 500/500 [00:09<00:00, 52.81it/s, bce_dice_loss - 1.414, iou_score - 0.6348]

Epoch: 29
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.488, iou_score - 0.5315]
valid: 100%|████| 500/500 [00:09<00:00, 52.74it/s, bce_dice_loss - 1.415, iou_score - 0.6337]

Epoch: 30
train: 100%|████| 396/396 [00:56<00:00,  7.59it/s, bce_dice_loss - 1.489, iou_score - 0.5301]
valid: 100%|████| 500/500 [00:09<00:00, 52.78it/s, bce_dice_loss - 1.415, iou_score - 0.6661]
Model saved!

Epoch: 31
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.487, iou_score - 0.5318]
valid: 100%|████| 500/500 [00:09<00:00, 52.71it/s, bce_dice_loss - 1.413, iou_score - 0.6402]

Epoch: 32
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.486, iou_score - 0.5344]
valid: 100%|█████| 500/500 [00:09<00:00, 52.78it/s, bce_dice_loss - 1.414, iou_score - 0.638]

Epoch: 33
train: 100%|████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.489, iou_score - 0.5308]
valid: 100%|███████| 500/500 [00:09<00:00, 52.93it/s, bce_dice_loss - 1.413, iou_score - 0.63

Epoch: 34
train: 100%|███████| 396/396 [00:56<00:00,  7.62it/s, bce_dice_loss - 1.486, iou_score - 0.53
valid: 100%|███████| 500/500 [00:09<00:00, 53.03it/s, bce_dice_loss - 1.413, iou_score - 0.64

Epoch: 35
train: 100%|███████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.487, iou_score - 0.53
valid: 100%|█████████| 500/500 [00:09<00:00, 52.59it/s, bce_dice_loss - 1.414, iou_score - 0.

Epoch: 36
train: 100%|███████| 396/396 [00:56<00:00,  7.59it/s, bce_dice_loss - 1.486, iou_score - 0.53
valid: 100%|███████| 500/500 [00:09<00:00, 52.71it/s, bce_dice_loss - 1.413, iou_score - 0.63

Epoch: 37
train: 100%|████████| 396/396 [00:56<00:00,  7.58it/s, bce_dice_loss - 1.487, iou_score - 0.5
valid: 100%|███████| 500/500 [00:09<00:00, 52.64it/s, bce_dice_loss - 1.413, iou_score - 0.63

Epoch: 38
train: 100%|███████| 396/396 [00:56<00:00,  7.61it/s, bce_dice_loss - 1.487, iou_score - 0.53
valid: 100%|███████| 500/500 [00:09<00:00, 52.55it/s, bce_dice_loss - 1.412, iou_score - 0.63

Epoch: 39
train: 100%|███████| 396/396 [00:56<00:00,  7.59it/s, bce_dice_loss - 1.486, iou_score - 0.53
valid: 100%|███████| 500/500 [00:09<00:00, 52.57it/s, bce_dice_loss - 1.412, iou_score - 0.63
Do you have any suggestions？

**Comment by sunxia233** [comment]
I re-executed the experiment, iou can reach 0.83

**Comment by sunxia233** [comment]
I ran the experiment on veri-776 according to your settings,
Accuracy obtained for the first time
mAP: 77.4%
[I 210121 15:20:54 main:473] CMC curve, Rank-1 :95.89%
[I 210121 15:20:54 main:473] CMC curve, Rank-5 :98.09%
[I 210121 15:20:54 main:473] CMC curve, Rank-10 :98.81%
[I 210121 15:20:54 main:338] Saving models in epoch 120
1.086038589477539

Accuracy obtained the second time:
mAP: 77.5%
[I 210116 11:03:52 main:475] CMC curve, Rank-1 :95.65%
[I 210116 11:03:52 main:475] CMC curve, Rank-5 :97.85%
[I 210116 11:03:52 main:475] CMC curve, Rank-10 :98.99%
[I 210116 11:03:52 main:337] Saving models in epoch 120
1.3036835193634033

Compared with the map in your paper, there is a drop of 2 points. What should I do to improve the accuracy, or is it because of the settings?

**Comment by silverbulletmdc** [comment]
I have fixed a bug recently. Please run `git pull origin master` to get the latest code and retrain it.
